scene/gui/tab_bar.cpp

```diff
@@ -998,7 +1021,7 @@ void TabBar::_update_cache(bool p_update_hover) {
 		tabs.write[i].size_text = Math::ceil(tabs[i].text_buf->get_size().x);
 		tabs.write[i].size_cache = get_tab_width(i);
 
-		if (max_width > 0 && tabs[i].size_cache > max_width) {
+		if ((tabs.write[i].truncated = max_width > 0 && tabs[i].size_cache > max_width)) {
```

Comment:
```suggestion
		tabs.write[i].truncated = max_width > 0 && tabs[i].size_cache > max_width;
		if (tabs[i].truncated) {
```

This should be split into two lines as mentioned above. https://github.com/godotengine/godot/pull/89247#discussion_r1518476658